### PR TITLE
README updates related to recent development script changes

### PR DESCRIPTION
**Pull request by dmohs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,77 +21,44 @@ After responding to changes, merge in GitHub.
...
-npm run start [-- --environment test]
+./project.rb dev_up
 ```
 
 After webpack finishes the build, you can view your local UI server at
-http://localhost:4200/.
+http://localhost:4200/. You can view the tests at http://localhost:9876/debug.html.
 
-By default, this connects to a local API server. Use `--environment=$ENV` to
+By default, this connects to a test API server. Use `--environment=$ENV` to
```
Comment (danqrodney) [inline README.md]: I don't think we want our local instances connecting to the test API server... do we?

File: README.md [hunk 2]
```diff
@@ -21,77 +21,44 @@ After responding to changes, merge in GitHub.
 
 ## Setup
 
-### API
-
-* Install JDK 8 on your machine from http://www.oracle.com/technetwork/java/javase/downloads/index.html.
-* Set your JAVA_HOME environment variable to point at the installation directory.
+Docker must be installed to build and run code.
```
Comment (danqrodney) [inline README.md]: What about Ruby?

File: README.md [hunk 3]
```diff
@@ -21,77 +21,44 @@ After responding to changes, merge in GitHub.
...
 use an alternate `src/environments/environment.$ENV.ts` file and connect to a
 different API server.
 
 #### You can regenerate classes from swagger with
 
 ```Shell
-npm run codegen
-```
-
-#### You can build the UI with
```
Comment (danqrodney) [inline README.md]: Why is this removed?

File: README.md (hunk 2 above)
Comment (markfickett) [inline README.md]: Could you add: "(For Google workstations, see go/installdocker.)"

File: README.md [hunk 4]
```diff
@@ -21,77 +21,44 @@ After responding to changes, merge in GitHub.
...
 
 ## Running the Dev Servers
 
 ### API: dev AppEngine appserver
 
 From the `api/` directory:
-
 ```Shell
-./gradlew appengineRun
+./project.rb dev_up
```
Comment (markfickett) [inline README.md]: The Docker container for the API is not quite done, right -- this won't work yet? I'm OK with changing it now if you expect we'll have the API in Docker shortly and prefer to batch up the README changes, just checking that I know what's going on.

File: README.md [hunk 5]
```diff
@@ -21,77 +21,44 @@ After responding to changes, merge in GitHub.
...
 local API server under http://localhost:8081/api/.
 
 ### UI
 
-From the `ui/` subdirectory:
-
+From the `ui/` directory:
 ```Shell
-npm run start [-- --environment test]
+./project.rb dev_up
```
Comment (markfickett) [inline README.md]: This still takes the optional `--environment [local|test|prod]` right?

Comment (markfickett) [inline README.md]: When I try to run it, `dev-up` is recognized (but not `dev_up`).

File: README.md (hunk 1 above)
Comment (markfickett) [inline README.md]: There was some discussion on the UI Dockerization PR. I like it because you don't have to start a local API if you don't need to test API changes.

File: README.md (hunk 2 above)
Comment (dmohs) [inline README.md]: Good call.

File: README.md (hunk 1 above)
Comment (dmohs) [inline README.md]: And if you don't mind an anecdote, I usually make UI changes without running the API server, so this fits my own use case quite well.

File: README.md (hunk 3 above)
Comment (dmohs) [inline README.md]: This command is executed prior to building the UI, but `./project.rb swagger-regen` runs this by itself. I added the help command (a bare `./project.rb`) to let developers more easily discover other available operations like this one.

File: README.md (hunk 4 above)
Comment (dmohs) [inline README.md]: I screwed up the deployment, so it isn't connecting to the MySQL database correctly when deployed, but the local spin-up should be working perfectly right now. Brian and I have both tested it, but it would be awesome to have a Chromebook test.

File: README.md (hunk 5 above)
Comment (dmohs) [inline README.md]: Yes. See `./project.rb dev-up --help`

Comment (dmohs) [inline README.md]: Oops! Thank you!

File: README.md (hunk 1 above)
Comment (danqrodney) [inline README.md]: So if you're making UI only changes, this seems fine.

If you're making API changes and want to make sure that the UI still works as-is, or if you're making both UI and API changes in the same PR, it isn't -- we need to run a local API server.

We should definitely make both simple; I just kind of liked having connection to your local server be the default, since it should always work (whether or not you're making API changes.)

Comment (danqrodney) [inline README.md]: Note: it's not clear how you're supposed to connect to the local API server from these directions. Either way, please fix that.

File: README.md (hunk 5 above)
Comment (markfickett) [inline README.md]: Should it be summarizing flags available to `dev-up`?
```
/ui$ ./project.rb --help

Usage: ./project.rb <command> <options>

COMMANDS

dev-up
Brings up the development environment.

install-dependencies
Installs dependencies via npm.

swagger-regen
Regenerates API client libraries from Swagger definitions.

clean-git-hooks
Removes symlinks created by shared-git-hooks. Necessary before re-installing.
```

File: README.md (hunk 1 above)
Comment (dmohs) [inline README.md]: Solid point. Fixed.
